packages/tc-markdown-parser/lib/parse-multiline-definition.js

```diff
@@ -0,0 +1,176 @@
+class SyntaxError extends Error {}
+
+/*
+ This function divides from mutipile definition to code and property string e.g,
+
+ The definition:
+ - child-code
+   propNameOne: propValueOne
+
+ Divides to:
+ {
+   code: 'child-code',
+   propDef: '  propNameOne: propValueOne\n',
+ }
+*/
+const dividePropertyDefinition = subdocument => {
+	const linefeedIndex = subdocument.indexOf('\n');
+	if (linefeedIndex > -1) {
+		return {
+			code: subdocument.slice(0, linefeedIndex),
+			propDef: subdocument.slice(linefeedIndex + 1),
+		};
+	}
+
+	return {
+		code: subdocument,
+		propDef: '',
+	};
+};
+
+// Token characters of meaningful property descriptor
+const TAB = '\t';
+const PROPERTY_NAME_SEPARATOR = ':';
+const LINE_FEED = '\n';
+const CARRIAGE_RETURN = '\r';
+
+/*
+ This is tiny lexer/parser fuction for analyze multiple property definition.
+ See above constant characters and parse into an object with following format e.g,
+
+ Definition
+   propNameOne: propValueOne
+   propNameTwo: propValueTwo
+ Will be parsed as
+   {
+     propNameOne: 'propValueOne',
+     propNameTwo: 'propValueTwo',
+   }
+*/
+const parsePropertyDefinition = (propDefString, line) => {
+	const properties = {};
+	const strlen = propDefString.length;
+	let buffer = '';
+	let propName = '';
+	let column = 0;
+
+	for (let i = 0; i < strlen; i++) {
```

Comment:
This could be replaced with a `for (const char of propDefString) {`, do you think that would be a good change to make?

Reply:
With thinking of loop performance, I chose primitive for-loop but I didn't realize ES6 syntax for-of loop. it looks neat! I'll use this, thanks!